File: 921-minimum-add-to-make-parentheses-valid/921-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {                                                                   int open = 0, close = 0, count = 0;
        for (auto &ch : s) {
            if (ch == '(') {
                count++;
            } else if (ch == ')') {
                if (count > 0) {
                    count--;
                } else {
                    close++;
                }
            }
        }
        return close + count;
    }
};

File: 921-minimum-add-to-make-parentheses-valid/921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        close = 0
        open = 0
        for char in s:
            if char == ')':
                if open > 0:
                    open -= 1
                else:
                    close += 1
            elif char == '(':
                open += 1
        return close + open